### PR TITLE
Restructured modal display and implemented sign in modal

### PR DESCRIPTION
**Pull request by onyi2020**
Kindly review when you are free and let me know if there are any changes that I should work on. Thank you

### CODE REVIEW

File: src/components/modal/login.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            border-radius: 0.5rem;
+            background-color: $darkgrayishBlue;
+            margin: 1rem 0;
+            &:focus {
+                outline: none;
+                background-color: $lightGrayishBlue;
+                border-bottom: $dark;
+                transition: background-color 0.5s ease-in-out;
+                + label {
+                    transform: translate(-20%, -100%) scale(0.75);
```
Comment (lape15) [inline src/components/modal/login.scss:26]: Nice! I never use + element  @onyi2020

File: src/components/modal/login.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    
+    return (
+        <div className='signInWrapper w-100' onClick={(e) => {e.stopPropagation()}}>
+            <div className='signInCredentials'>
+                <h3>Sign in</h3>
+                <hr />                
+                <div className='input-container'>
+                    <input type='text' id='username' className={`p-1 w-100 ${className ? 'is-valid' : ''}`} 
+                    onBlur={(e) => handleBlur(e)}></input>
+                    <label htmlFor='username' className=''>Username</label>
```
Comment (lape15) [inline src/components/modal/login.tsx:19]: What if we want to run validation checks? @onyi2020

**Review by lape15 — APPROVED** [review]
Looks good @onyi2020, can I merge?

File: src/components/modal/login.scss (hunk 1 above)
Comment (onyi2020) [reply src/components/modal/login.scss:26]: Any reason why?

File: src/components/modal/login.tsx (hunk 2 above)
Comment (onyi2020) [reply src/components/modal/login.tsx:19]: Whatkind of validation checks?

**Comment by onyi2020** [comment]
If you think it is good then merge. I refactor for thevalidation check you talke about in the comment above. We will talk about the backend on whatsapp

**Comment by onyi2020** [comment]
I have just added more commits..you might have to review new commit before finally merging

File: src/components/image-card/image-card.tsx [hunk 3]
```diff
@@ -1,55 +1,18 @@
...
 
   const setActiveView = (index: number) => {
-    const temp = [...shoeView].map((sh, idx) => ({
-      ...sh,
-      active: index === idx ? true : false,
-    }));
-    setCardView(temp[index]);
-    setShoeView([...temp]);
+    dispatch(getActiveProduct(index))
+    console.log(cardView.main, {cardView})
```
Comment (lape15) [inline src/components/image-card/image-card.tsx]: remove console.log @onyi2020

File: src/components/modal/product-modal.tsx [hunk 4]
```diff
@@ -1,52 +1,58 @@
...
+    
+    const getActiveItem = (shop: typeof shoeView) => {
+        let itemIndex;
+        for(const sh in shop){
+            const item = shop[sh]
+            if(item.active === true){
+                itemIndex = sh;
+            }
+        }
+        return itemIndex;
```
Comment (lape15) [inline src/components/modal/product-modal.tsx:20]: Did typescript complain about the type of itemIndex? @onyi2020

File: src/components/modal/product-modal.tsx [hunk 5]
```diff
@@ -1,52 +1,58 @@
...
+    }
+
+    const doNext = () => {
+        let index: any = getActiveItem(shoeView)
+        if(index && index >= 0 && index < shoeView.length-1){
```
Comment (lape15) [inline src/components/modal/product-modal.tsx:25]: why any??? @onyi2020

File: src/components/modal/login.tsx (hunk 2 above)
Comment (lape15) [reply src/components/modal/login.tsx:19]: I just meant in the future, if we want to check against what a user types @onyi2020

Comment (onyi2020) [reply src/components/modal/login.tsx:19]: Maybe we can do it inside the handleBlur function or when submt is triggered

File: src/components/modal/product-modal.tsx (hunk 4 above)
Comment (onyi2020) [reply src/components/modal/product-modal.tsx:20]: Nope.

File: src/components/modal/product-modal.tsx (hunk 5 above)
Comment (onyi2020) [reply src/components/modal/product-modal.tsx:25]: I added a comment regarding this.. index should be a number but I got typescript error, not sure how to convert it to number

File: src/components/image-card/image-card.tsx (hunk 3 above)
Comment (onyi2020) [reply src/components/image-card/image-card.tsx]: Done

File: src/components/modal/login.tsx (hunk 2 above)
Comment (lape15) [reply src/components/modal/login.tsx:19]: ohk @onyi2020

File: src/components/modal/product-modal.tsx (hunk 5 above)
Comment (lape15) [reply src/components/modal/product-modal.tsx:25]: Weird

**Comment by lape15** [comment]
I have merged @onyi2020

**Pull request merged**